packages/visualizations/src/components/Map/WebGl/MapRender.svelte

```diff
@@ -94,30 +93,22 @@
     const sourceId = `shape-source-${mapId}`;
     const layerId = `shape-layer-${mapId}`;
 
-    const fitBox = (box: BBox | LngLatBoundsLike) => {
-        // Using padding, keep enough room for controls (zoom) to make sure they don't hide anything
-        map.fitBounds(box as LngLatBoundsLike, {
-            animate: false,
-            padding: 40,
-        });
-    };
-
-    const setViewBox = (box: BBox | undefined) => {
+    const setBbox = (box: BBox | undefined) => {
         if (!box) {
             // zoom-out to bounds defined in the initialization
             map.setZoom(map.getMinZoom());
             return;
         }
-        fitBox(box);
-    };
-
-    function setMaxBounds(bounds: BBox) {
         // Cancel any saved max bounds to properly fitBounds
         map.setMaxBounds(null);
-        fitBox(bounds);
-        // Reset min zoom and movement
+        // Using padding, keep enough room for controls (zoom) to make sure they don't hide anything
+        map.fitBounds(box as LngLatBoundsLike, {
+            animate: false,
+            padding: 40,
+        });
+        // Set new map max bounds after bbox changes
         map.setMaxBounds(map.getBounds());
```

Comment:
I'm not sure I get your point here, if we don't undo the `maxBounds` here we can"t `fitBounds` properly so we can't then `getBounds` and adapt `setMaxBounds`... 

All this bounds are a bit confusing but I don't see a clear another solution here if you want to make the map automatically resize, sometimes change the map canvas ratio and still properly fit the map bounds and restrict the bounds than just do all the actions when the map canvas changes. Splitting all those actions seemed to lead to many bugs as they are not really independent and seem all useful in our case. setMaxBounds should adapt to the new bbox but also to the new size of the map canvas. So unless I'm missing something I don't see how you can achieve that separately but I'm not a Svelte expert and there might be a way to do that differently, if you have some tips here it would be great 😃